### ISSUE TITLE
No internet

### ISSUE THREAD

**Issue by jdavid82** [pivpn/pivpn#76]
N00b here, after successful installation the raspberry pi could no longer connect to the internet, please advice, thanks.

**Comment by guydeke** [comment]
That's not enough information, only no access with PI? Or with your network? Do you use a firewall? Is DNS configured well? Does your router block VPN trafic? ....... and thousend questions more :)

**Comment by jdavid82** [comment]
You are right, not enough info, I thought it was a common problem, I set it up last night and I was able to connect from a windows machine tethered from my 3G phone, I even ran a node server on it and it all worked well, however I had some source code in the raspberry pi and need to do a git pull and it said: fatal: unable to access 'https://github.com/blahblah... then I realised I couldn't ping google.com either.
I uninstalled pivpn and rebooted and still can't connect, but that's most likely because I broke it while going through forums and stack overflow questions and doing stupid things such as running random commands that people suggest without taking the time to actually read through the documentation.
Unfortunately I don't have much time, if I have more time in the next few weeks then I'll format the sd card and start from scratch and give it another go, in the mean time I'll disable port forwarding on my router and go do some other stuff I have to work on. Thank you, I'll keep an eye on it especially if someone else had this issue too.

**Comment by jdavid82** [comment]
oh and I was using wlan0 instead of eth0, I don't know if pivpn scripts would take that into account

**Comment by guydeke** [comment]
I don't know, but maybe you have a DNS problem.
You can change the DNS in the Google dns by typing from command line:

nano /etc/resolv.conf

# Then add the two lines below:
nameserver 8.8.8.8
nameserver 8.8.4.4

But I don't know if it solves your problem.

**Comment by jdavid82** [comment]
Thank you, I will format the SD card and re-install the OS and pivpn and will give it another go, will post my findings here.

**Comment by jdavid82** [comment]
OK I re-installed the OS along with pivpn and again lost connection to the internet, then I followed your suggestion and after restarting the contents of /etc/resolv.conf went back to its original state (I did make sure that the changes were being saved). Its contents are as follows:
`# Generated by resolvconf
nameserver 192.168.0.1
`
I searched in Google for an answer and saw a few different "possible solutions", however I decided to add my comment on here first and see if you had any further suggestions, thanks.

**Comment by guydeke** [comment]
It looks that now you called your gateway a DNS, that's not good!
Read about the Google DNS (that will do the DNS job!) here: https://en.wikipedia.org/wiki/Google_Public_DNS

so you should change the nameserver in 8.8.8.8 or 8.8.4.4

nameserver 8.8.8.8
nameserver 6.6.4.4

You can put both in your file.
I don't now if this solves your problem, but your nameserver can't be your gateway.

**Comment by jdavid82** [comment]
Ok got it working, I ssh-ed into the device and:
`sudo vim /etc/network/interfaces`
And under `iface wlan0` (because I'm using the wireless antenna I guess), I added this entry:
`dns-nameservers 8.8.8.8 8.8.4.4`
then `:x` to save and quit and then:
`sudo shutdown -r now` to reboot.
Next time I ssh-ed in I was able to connect to the internet.
Thank you for your help and I hope this helps someone else.
